speechbrain/lobes/models/huggingface_transformers/speecht5.py

```diff
@@ -0,0 +1,483 @@
+        del self.model
+        gc.collect()
+        if torch.cuda.is_available():
+            torch.cuda.empty_cache()
+
+    def get_generation_config(self) -> GenerationConfig:
+        """Returns the HuggingFace Transformers generation config associated with SpeechT5.
+        If the generation config is None, it will create a new one.
+
+        Returns
+        -------
+        GenerationConfig
+            Model generation config.
+        """
+        if self.generation_config is None:
+            self.generation_config = GenerationConfig(
+                decoder_start_token_id=2,
+                eos_token_id=self.model.config.eos_token_id,
+                pad_token=self.model.config.pad_token_id,
+                max_length=450,
+            )
+
+        return self.generation_config
+
+    @mark_as_loader
+    def load_checkpoint_hf(self, path, end_of_epoch) -> None:
+        """Custom checkpoint loading hook used to avoid the model size mismatch when using Pytorch's format.
+        This function will call the transformers library implementations of `from_pretrained()` to load :
+            - The model
+            - The model configuration
+            - The generation configuration
+
+        Arguments
+        ----------
+        path : os.PathLike or str
+            Path of the checkpoint directory.
+        end_of_epoch : bool
+            Whether or not the checkpoint was saved duraing the end of an epoch.
+        """
+        self._free_memory()
+        self.config = SpeechT5Config.from_pretrained(path)
+
+        if torch.cuda.is_available():
+            self.model = SpeechT5ForSpeechToText.from_pretrained(
+                path,
+                config=self.config,
+                local_files_only=True,
+                ignore_mismatched_sizes=True,
+                use_safetensors=True,
+            ).cuda()
+        else:
+            self.model = SpeechT5ForSpeechToText.from_pretrained(
+                path,
+                config=self.config,
+                local_files_only=True,
+                ignore_mismatched_sizes=True,
+                use_safetensors=True,
+            )
+
+        self.generation_config = GenerationConfig.from_pretrained(
+            path, "generation_config.json"
+        )
+
+    @mark_as_saver
+    def save_checkpoint_hf(self, path):
+        """Custom checkpoint saving hook used to avoid the model size mismatch when using Pytorch's format.
+        This function will call the transformers library implementationsof `save_pretrained()` to save :
+            - The model and its configuration
+            - The generation configuration
+
+        Arguments
+        ----------
+        path :
+            Path where the checkpoint will be saved
+        """
+        self.model.save_pretrained(
+            path, from_pt=True, state_dict=self.model.state_dict()
+        )
+        self.get_generation_config().save_pretrained(
+            path, "generation_config.json"
+        )
```

Comment:
I think this part is a bit sus to me. Can you please show me what is exactly happening with the shape mismatch? Ty my dude